File: 프로그래머스/1/12982. 예산/예산.swift
import Foundation

func solution(_ d:[Int], _ budget:Int) -> Int {

    let sortedArr = d.sorted()
    
    var ans = 0
    var result = 0
    for i in 0..<sortedArr.count {
        result += sortedArr[i]
        if result > budget {
            ans = i
            break
        }
    }
    if result <= budget {
        ans = sortedArr.count
    }
    return ans
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        int ans = 0;
        int[] prefixSum = new int[d.length + 1];
        Arrays.sort(d);
        for(int i=1; i<=d.length; i++) {
            prefixSum[i] = prefixSum[i-1] + d[i-1];
        }
        
        for(int i=1; i<= d.length; i++) {
            if(prefixSum[i] <= budget) {
                ans = i;
            } else { break;}
        }
        
        
        return ans;
    }
}